plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/internal/N4JSEclipseModel.java

```diff
@@ -128,12 +128,28 @@ public N4JSEclipseProject findProjectWith(URI nestedLocation) {
 
 				if (location.isFile()) {
 					final IN4JSEclipseProject eclipseProject = findProjectWith(location);
-					if (null != eclipseProject && eclipseProject.exists()) {
+					if (eclipseProject != null && eclipseProject.exists()) {
```

Comment:
whole `org.eclipse.n4js.ui.internal.N4JSEclipseModel.findN4JSSourceContainer(URI)` needs refactoring.

Reply:
Yes. I just changed some minor issues when I read the code and tried to understand it.